examples/arduino_keyboard.rs

```diff
@@ -145,6 +144,20 @@ impl UsbContext {
                 }
             }
         }
+
+        // Keyboard reports can be sent at max 1/ms.
+        // Wait some cycles as a simple debouncer.
+        delay_ms(5);
+    }
+
+    fn delay_repeated(&self) {
+        // Add an extra delay for repeated keys.
+        //
+        // The actual delay time necessary could vary by OS, and/or config settings on the same OS. TBD
+        if self.current_index > 0 && PAYLOAD[self.current_index - 1] == PAYLOAD[self.current_index]
+        {
+            delay_ms(600);
+        }
```

Comment:
These delays aren't the best solution, and adding a blocking delay is not acceptable for this example. In this case, it's being called from an interrupt context, where you want to handle interrupts as quickly as possible. But even if it was in the main thread, it would still completely block that thread and prevent anything else from running.

You don't need to add delays to accommodate a USB host. The host is going to attempt to read reports however quickly it can handle the reports (while also respecting the minimum interval, set by the device in `HIDClass::new`)

As I mentioned in #5 , the correct way to fix this is to send an empty report after each character, so you send a single report for each "press" and "release" of each character.

Reply:
I'll try to get that working. So far, the only thing I could do to get it to work was the long delay. Literally anything less than 600ms results in the host eating repeated characters.

Reply:
Just to make sure, are you checking the result of `push_input()`? It needs to be `Ok` before sending the next report, otherwise it hasn't been successfully sent to the host yet.